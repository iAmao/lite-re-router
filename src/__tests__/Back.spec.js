import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';

import { Back } from '../';


describe('Back', () => {
    describe('Snapshot Test', () => {
        it('renders a link', () => {
            const tree = renderer.create(
                <Back>Button</Back>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('Unit Test', () => {
        it('should change context on click', () => {
            let count = 0;
            const wrapper = shallow(Back(
                { children: 'Button' },
                { location: { back: () => count++}}
            ));

            wrapper.simulate('click');

            expect(count).toEqual(1);
            expect(wrapper.text()).toEqual('Button');
        })
    });
});
