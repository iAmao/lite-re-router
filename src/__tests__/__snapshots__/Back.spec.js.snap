// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Snapshot Test renders a link 1`] = `
<a
  onClick={[Function]}
  style={
    Object {
      "color": "blue",
      "cursor": "pointer",
      "textDecoration": "underline",
    }
  }
>
  Button
</a>
`;
